lib/ramble/ramble/software_environments.py

```diff
@@ -20,402 +20,653 @@
 import ramble.util.matrices
 import ramble.util.colors as rucolor
 
+package_managers = Enum("package_managers", ['spack'])
 
-class SoftwareEnvironments(object):
-    """Class to represent a set of software environments
 
-    This class contains logic to take the dictionary representations of
-    software environments, and unify their format.
-    """
+class SoftwarePackage(object):
+    """Class to represent a single software package"""
 
-    supported_confs = ['v2']
+    def __init__(self, name: str, spec: str, compiler: str = None,
+                 compiler_spec: str = None, package_manager=None):
+        """Software package constructor
 
-    def __init__(self, workspace):
-        self.keywords = ramble.keywords.keywords
-
-        self._raw_packages = {}
-        self._packages = {}
-        self._package_map = {}
-        self._raw_environments = {}
-        self._environments = {}
-        self._environment_map = {}
-        self._workspace = workspace
-        self.spack_dict = self._workspace.get_spack_dict().copy()
+        Args:
+            name (str): Name of package
+            spec (str): Package spec (used to install / load package)
+            compiler (optional str): Name of package definition to use as compiler
+                                     for this package
+            compiler_spec (optional str): Spec string to use when this package
+                                          is used as a compiler
+            package_manager (optional str): Name of package manager for this package
+        """
 
-        conf_type = self._detect_conf_type()
+        self.name = name
+        self.spec = spec
+        self.compiler = compiler
+        self.compiler_spec = compiler_spec
+        self._package_type = 'Rendered'
+
+        if package_manager and hasattr(package_managers, package_manager):
```

Comment:
Do you think this would ber cleaner with something more like 

```
self.package_manager = package_managers[package_manger]
```

In what situations are we expecting to get passed an invalid package manager and still want to continue?

Reply:
Yeah, that's a good point. I think we generally don't want to allow "unsupported" package managers. So it probably makes sense to just error here.